src/main/scala/cromwell/engine/callexecution/CallExecutionActor.scala

```diff
@@ -18,43 +21,27 @@ object CallExecutionActor {
   final case class PollResponseReceived(executionHandle: ExecutionHandle) extends CallExecutionActorMessage
   final case class Finish(executionHandle: ExecutionHandle) extends CallExecutionActorMessage
 
-  sealed trait ExecutionMode extends CallExecutionActorMessage {
-    def execute(backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle]
-  }
-
-  case object Execute extends ExecutionMode {
-    override def execute(backendCall: BackendCall)(implicit ec: ExecutionContext) = backendCall.execute
-  }
+  sealed trait ExecutionMode extends CallExecutionActorMessage
 
-  final case class Resume(jobKey: JobKey) extends ExecutionMode {
-    override def execute(backendCall: BackendCall)(implicit ec: ExecutionContext) = backendCall.resume(jobKey)
-  }
+  case object Execute extends ExecutionMode
+  final case class Resume(jobKey: JobKey) extends ExecutionMode
+  final case class UseCachedCall(cachedBackendCall: BackendCall) extends ExecutionMode
 
-  final case class UseCachedCall(cachedBackendCall: BackendCall, backendCall: BackendCall) extends ExecutionMode {
-    override def execute(backendCall: BackendCall)(implicit ec: ExecutionContext) = backendCall.useCachedCall(cachedBackendCall)
-  }
-
-  def props(backendCall: BackendCall): Props = Props(new CallExecutionActor(backendCall))
+  def props(backendCall: BackendCall): Props = Props(new BackendCallExecutionActor(backendCall))
+  def props(finalCall: FinalCall): Props = Props(new FinalCallExecutionActor(finalCall))
 }
 
-/** Actor to manage the execution of a single call. */
-class CallExecutionActor(backendCall: BackendCall) extends Actor with CromwellActor {
-  import CallExecutionActor._
+trait CallExecutionActor extends Actor  with CromwellActor {
 
-  val akkaLogger = Logging(context.system, classOf[CallExecutionActor])
-  val logger = WorkflowLogger(
-    "CallExecutionActor",
-    backendCall.workflowDescriptor,
-    akkaLogger = Option(akkaLogger),
-    callTag = Option(backendCall.key.tag)
-  )
+  def akkaLogger = Logging(context.system, classOf[CallExecutionActor])
```

Comment:
I've gone back and forth on how I personally feel about this but I think the general stance is that if you feel strongly enough about a default implementation that it should actually be a default implementation that using val is fine